Public/Src/Utilities/KeyValueStore/KeyValueStoreAccessor.cs

```diff
@@ -89,10 +114,25 @@ public partial class KeyValueStoreAccessor : IDisposable
         public bool ReadOnly { get; }
 
         /// <summary>
-        /// Called when an exception is thrown, allows for custom exception handling.
+        /// Determines when the failure handler will be called, when the store will be locked down because of an 
+        /// error, and whether an exception should be rethrown.
+        /// </summary>
+        private readonly Func<Failure<Exception>, (bool lockdown, bool rethrow)> m_errorPolicy;

```

Comment:
Instead of using a tuple, I would suggest to wrap this thing into a separate type, similar to UnobservedTaskExceptionEventArgs ([https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.unobservedtaskexceptioneventargs?view=netframework-4.8](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.unobservedtaskexceptioneventargs?view=netframework-4.8)).

To be honest, I'm not a big fan of using tuples in the public surface.